### ISSUE TITLE
How to suppress warnings in svelte with sveltekit?

### ISSUE THREAD

**Issue by jonathanblancor** [sveltejs/language-tools#1079]
I understand that warnings are disabled when running `svelte-check` command (with `--compiler-warnings css-unused-selector:ignore,unused-export-let:error`). 

Now, is there a way to integrate this when running `svelte-kit dev` and disable the warnings in the VS Code console in development?

**Comment by dummdidumm** [comment]
Related to #855 . If this is adopted and all tools use that property, this would be the way.

**Comment by jonathanblancor** [comment]
Thanks! I'm closing this one then.

**Comment by craigcosmo** [comment]
To disable almost all the blue warning in sveltekit, do this

open setting.json in vscode and add this

```json
"svelte.plugin.svelte.compilerWarnings": {
		"css-unused-selector": "ignore",
		"unused-export-let": "ignore",
		"a11y-aria-attributes": "ignore",
		"a11y-incorrect-aria-attribute-type": "ignore",
		"a11y-unknown-aria-attribute": "ignore",
		"a11y-hidden": "ignore",
		"a11y-autocomplete-valid": "ignore",
		"a11y-misplaced-role": "ignore",
		"a11y-no-static-element-interactions": "ignore",
		"a11y-unknown-role": "ignore",
		"a11y-no-abstract-role": "ignore",
		"svelte-ignore a11y-autofocus": "ignore",
		"a11y-no-redundant-roles": "ignore",
		"a11y-role-has-required-aria-props": "ignore",
		"a11y-accesskey": "ignore",
		"a11y-autofocus": "ignore",
		"a11y-misplaced-scope": "ignore",
		"a11y-positive-tabindex": "ignore",
		"a11y-invalid-attribute": "ignore",
		"a11y-missing-attribute": "ignore",
		"a11y-img-redundant-alt": "ignore",
		"a11y-label-has-associated-control": "ignore",
		"a11y-media-has-caption": "ignore",
		"a11y-distracting-elements": "ignore",
		"a11y-structure": "ignore",
		"a11y-mouse-events-have-key-events": "ignore",
		"a11y-missing-content": "ignore",
		"a11y-click-events-have-key-events": "ignore",
		"a11y-no-noninteractive-element-interactions": "ignore"
	},
```

Then open the svelte.config.js file and do this

```js
const config = {
	preprocess: vitePreprocess(),
	onwarn: (warning, handler) => {
		if (warning.code.startsWith('a11y-')) return
		if (warning.code === 'missing-exports-condition') return
		if (warning.code === 'a11y-no-static-element-interactions') return
		if (warning.code === 'svelte-ignore a11y-autofocus') return
		if (warning.code.startsWith('css-unused-selector')) return
		handler(warning)
	},
	kit: {
		adapter: adapter(),
	},
}
```

Add more rules you hate into those files/config.

**Comment by dslatkin** [comment]
> To disable almost all the blue warning in sveltekit, do this
> 
> open setting.json in vscode and add this
> 
> ...

Awesome, thank you @craigcosmo. I should have just tried this but I was expecting an `onwarn` type to show up somewhere by default. Walked into this trying to build a component library that I intend to compile twice, once to Svelte components and once using the custom elements API, so I needed to disable the `missing-custom-element-compile-options` warning for the first compile (where that option is effectively ignored).

If you want to add type safety for this function, I found this works:

```js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  // ...

  /** @type {import('@sveltejs/vite-plugin-svelte').SvelteConfig['onwarn']} */
  onwarn: (warning, handler) => {
    // ...
  },
};

export default config;
```

Not sure if `onwarn` is intentionally excluded from `@type {import('@sveltejs/kit').Config}` implying it's not technically supported/guaranteed to exist in the future. Maybe FYI @dummdidumm in case it would make sense to add to that SvelteKit config type if it's okay to actually use this way...
